charms/kfp-api/src/charm.py

```diff
@@ -210,64 +207,63 @@ def _kfp_api_layer(self) -> Layer:
 
         return Layer(layer_config)
 
-    def _generate_config(self, interfaces):
-        """Generate configuration based on supplied data.
+    def _generate_environment(self) -> dict:
+        """Generate environment based on supplied data.
 
         Configuration is generated based on:
         - Supplied interfaces.
         - Database data: from MySQL relation data or from data platform library.
         - Model configuration.
+
+        Return:
```

Comment:
Good one, we should always try to have a proper docstring in our functions. :clap: